x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/rest/action/user/RestHasPrivilegesActionTests.java

```diff
@@ -38,6 +40,7 @@
      */
     public void testBodyConsumed() throws Exception {
         final XPackLicenseState licenseState = mock(XPackLicenseState.class);
+        when(licenseState.isSecurityEnabled()).thenReturn(true);
```

Comment:
To retain the existing behaviour of this test, should this mock return `false` instead of `true` so that `action.handleRequest(...)` does not throw exception but returns a response about security is not enabled (currently it returns security not available due to license)?

Reply:
We discussed this in another channel. After another look, I think the change is in fact closer to what this test tries to assert:
1. Send a request without an username
2. It should _not_ fail with IllegalArgumentException complaining about unconsumed payload

Both the existing version and the change do the job because both of them will run through `BaseRestHandler#handleRequest`, where the unconsumed body is checked.
The existing version does its job by ensure there is no exception. This relies on the fact that “security disabled” message takes precedence over the “no authenticated user” exception. The new change enables security so that the “no authenticated user” exception now bubbles up. Both “security disabled” message and “no authenticated user” exception have lower priority than the “unconsumed body” exception. So the test does its job if either of them can be asserted. But the new change has the advantage of an additional assertion on the exception to be “no authenticated user”, which is more to the point.

In summary, the new change does its job slightly differently than the existing version. But I think it's a small improvement. So we can discard my previous comment.